source/common/http/filter/gzip_filter.cc

```diff
@@ -0,0 +1,274 @@
+  CONSTRUCT_ON_FIRST_USE(std::unordered_set<std::string>,
+                         {"text/html", "text/plain", "text/css", "application/javascript",
+                          "application/json", "image/svg+xml", "text/xml",
+                          "application/xhtml+xml"});
+}
+
+} // namespace
+
+GzipFilterConfig::GzipFilterConfig(const envoy::api::v2::filter::http::Gzip& gzip)
+    : compression_level_(compressionLevelEnum(gzip.compression_level())),
+      compression_strategy_(compressionStrategyEnum(gzip.compression_strategy())),
+      content_length_(contentLengthUint(gzip.content_length().value())),
+      memory_level_(memoryLevelUint(gzip.memory_level().value())),
+      window_bits_(windowBitsUint(gzip.window_bits().value())),
+      content_type_values_(contentTypeSet(gzip.content_type())),
+      disable_on_etag_header_(gzip.disable_on_etag_header()),
+      disable_on_last_modified_header_(gzip.disable_on_last_modified_header()) {}
+
+ZlibCompressionLevelEnum
+GzipFilterConfig::compressionLevelEnum(GzipV2CompressionLevelEnum compression_level) {
+  switch (compression_level) {
+  case GzipV2CompressionLevelEnum::Gzip_CompressionLevel_Enum_BEST:
+    return ZlibCompressionLevelEnum::Best;
+  case GzipV2CompressionLevelEnum::Gzip_CompressionLevel_Enum_SPEED:
+    return ZlibCompressionLevelEnum::Speed;
+  default:
+    return ZlibCompressionLevelEnum::Standard;
+  }
+}
+
+ZlibCompressionStrategyEnum
+GzipFilterConfig::compressionStrategyEnum(GzipV2CompressionStrategyEnum compression_strategy) {
+  switch (compression_strategy) {
+  case GzipV2CompressionStrategyEnum::Gzip_CompressionStrategy_RLE:
+    return ZlibCompressionStrategyEnum::Rle;
+  case GzipV2CompressionStrategyEnum::Gzip_CompressionStrategy_FILTERED:
+    return ZlibCompressionStrategyEnum::Filtered;
+  case GzipV2CompressionStrategyEnum::Gzip_CompressionStrategy_HUFFMAN:
+    return ZlibCompressionStrategyEnum::Huffman;
+  default:
+    return ZlibCompressionStrategyEnum::Standard;
+  }
+}
+
+std::unordered_set<std::string>
+GzipFilterConfig::contentTypeSet(const Protobuf::RepeatedPtrField<std::string>& types) {
+  // This is only called once by the constructor.
+  return types.empty() ? defaultContentEncoding()
+                       : std::unordered_set<std::string>(types.cbegin(), types.cend());
+}
+
+uint64_t GzipFilterConfig::contentLengthUint(Protobuf::uint32 length) {
+  return length >= MinimumContentLength ? length : MinimumContentLength;
+}
+
+uint64_t GzipFilterConfig::memoryLevelUint(Protobuf::uint32 level) {
+  return level > 0 ? level : DefaultMemoryLevel;
+}
+
+uint64_t GzipFilterConfig::windowBitsUint(Protobuf::uint32 window_bits) {
+  return (window_bits > 0 ? window_bits : DefaultWindowBits) | GzipHeaderValue;
+}
+
+GzipFilter::GzipFilter(const GzipFilterConfigSharedPtr& config)
+    : skip_compression_{true}, compressed_data_(), compressor_(), config_(config) {}
+
+FilterHeadersStatus GzipFilter::decodeHeaders(HeaderMap& headers, bool) {
```

Comment:
add comment on the bool arg indicating what it's supposed to be

Reply:
I see the point in adding a comment, but I looked at the other filters and none has comments for unused arguments. I add the argument name in the header. Should it be enough?

E.g.,

fault filter -> https://github.com/envoyproxy/envoy/blob/921530a0ac7e047890981467b9e44bb5de6321ef/source/common/http/filter/fault_filter.cc#L201